Testscripts/Linux/channel_change.sh

```diff
@@ -154,26 +168,113 @@ function Main() {
 							LogMsg "Successfully found dmesg log per cpu online state change"
 						else
 							LogErr "Failed to verify cpu online state change. Expected smpboot: Booting Node, but found $diff_val2"
+							FailedCount=$((FailedCount+1))
 						fi
 					else
 						LogErr "Failed to change back cpu $id to online"
+						FailedCount=$((FailedCount+1))
 					fi
 				else
 					LogErr "Failed to find the expected dmesg. Expected CPU $id is now offline, but found $diff_val"
+					FailedCount=$((FailedCount+1))
 				fi
 			else
 				LogErr "Failed to change the cpu $id state to offline. Expected 0, but found $post_state"
+				FailedCount=$((FailedCount+1))
 			fi
 		else
 			LogErr "Found the currect cpu $id is not online. Expected 1 but found $state"
+			FailedCount=$((FailedCount+1))
 		fi
 	done
 
+	# ########################################################################
+	# Assign a random CPU id to the vmbus channel
+	# Assign the original CPU id to the vmbus channel
+	# Now each vmbus's cpu id is 0.
+
+	# Find the max CPU core counts
+	max_cpu=$(nproc)
+	LogMsg "Re-read the original lsvmbus output file"
+	while IFS=' ' read -a line
+	do
+		if [[ ${line[0]} == "VMBUS" && ${line[1]} == "ID" ]]; then
+			vmbus_id[$idx]=${line[2]%?}
+			LogMsg "Found new VMBUS ID --> ${vmbus_id[$idx]}"
+
+			read -a line
+			# Ignored Device Id
+
+			read -a line
+			if [[ ${line[0]} =~ "Sysfs" && ${line[1]} == "path:" ]]; then
+				sysfs_path=${line[2]}
+				LogMsg "Found new sysfs --> $sysfs_path"
+
+				# Read Rel_ID and target_cpu lines
+				read -a line
+				while [[ $line != '' ]]; do
+					rel_id=${line[0]#"$prefix1"}
+					rel_id=${rel_id%"$suffix1"}
+					LogMsg "Found Rel_ID: $rel_id, which is vmbus channel id"
+					cpu_id=${line[1]#"$prefix2"}
+					LogMsg "Found the original target cpu id: $cpu_id"
+					# This cpu_is is fro lsvmbus.output, the original output.
```

Comment:
fro for or from?

Reply:
Fixed in 80240b93a6678b2c1f77ddd6f53163e4410bc0c8